spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/RegressionIdeas20090105Test.java

```diff
@@ -0,0 +1,37 @@
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.AbstractIntegrationTest;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcher;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
+import org.junit.jupiter.api.Test;
+
+import static edu.umd.cs.findbugs.test.CountMatcher.containsExactly;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItem;
+
+class RegressionIdeas20090105Test extends AbstractIntegrationTest {
+    @Test
+    void test() {
+        performAnalysis("bugIdeas/Ideas_2009_01_05.class");
+        assertBugNum(2);
+        assertExactBug("test", 25);
+        assertExactBug("test2", 29);
+    }
+
+    private void assertBugNum(int num) {
```

Comment:
I wonder if we shouldn't move these `assertXyzBug` methods in `AbstractIntegrationTest` because we seem to be duplicating them all over the place.
In theory they would only wrap `BugInstanceMatcher`'s fluent API but I think that it would be nicer to have built-in methods for the most frequent cases

Reply:
Yes, there are code a lot of code repetition with these asserts, but there are already a lot of similar code duplication. IMO it would be best if the already existing similar code would use these functions, but that would make this PR too big and messy, and would be out of scope of this PR. I'll create a new PR in a few days adding them, and after it's merged, make this PR using that feature.

Reply:
Sounds good to me, thank you for the improvements!

Reply:
Created https://github.com/spotbugs/spotbugs/pull/3120 to solve this issue.